pkgs/top-level/haskell-ng.nix

```diff
@@ -77,8 +81,9 @@ rec {
       packageSetConfig = callPackage ../development/haskell-modules/configuration-ghc-head.nix { };
     };
     ghcjs = callPackage ../development/haskell-modules {
+      ghc = compiler.ghcjs;
       packageSetConfig = callPackage ../development/haskell-modules/configuration-ghcjs.nix { };
+      jailbreak-cabal = "${packages.ghc7101.jailbreak-cabal}";
```

Comment:
Yes, that seems much cleaner to me. That would also address your other comment:

> We already have two mechanisms to replace jailbreak-cabal: the value can be overridden for the entire package set, and it can be overridden directly on mkDerivation. Adding a third way seems superfluous to me.

I'll make that change - thanks for the suggestion :).